### PR TITLE
test: CXTB-390 Automate Navigate to Participant and Access Participant Directory & Filter and Can Click and View Participant Record from Listing

### PR DESCRIPTION
**Pull request by cmoreno-nts**
Automated Test Case:
https://digitalscientists.atlassian.net/browse/CXTB-390

image

### CODE REVIEW

File: tests/cases/providers_care_platform/case_providers_participants.yaml [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+      App: "desktop"
+  Actions:
+  # Log in Care Platform as an Provider
+    - Type: case
+      Value: ProvidersCarePlatformLogin
+  # Go to Participants
+    - Type: click
+      Strategy: xpath
+      Id: $PAGE.care_platform.navigation_participant$      
+#CXTB-171: Access Participant Directory & Filter
```
Comment (dguzmanneural) [inline tests/cases/providers_care_platform/case_providers_participants.yaml:15]: Could you put this on top with the CXTB-170?

Comment (cmoreno-nts) [reply tests/cases/providers_care_platform/case_providers_participants.yaml]: understood, already fixed.

File: tests/cases/providers_care_platform/case_providers_participants.yaml [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+    - Role: desktopChrome
+      App: "desktop"
+  Actions:
+  # Log in Care Platform as an Provider
+    - Type: case
+      Value: ProvidersCarePlatformLogin
+  # Go to Participants
+    - Type: click
+      Strategy: xpath
+      Id: $PAGE.care_platform.navigation_participant$
```
Comment (csanchez-nts) [inline tests/cases/providers_care_platform/case_providers_participants.yaml:15]: This should be a provider page not a care platform

File: tests/cases/providers_care_platform/case_providers_participants.yaml [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+    - Type: case
+      Value: ProvidersCarePlatformLogin
+  # Go to Participants
+    - Type: click
+      Strategy: xpath
+      Id: $PAGE.care_platform.navigation_participant$      
+  # Get the value to compare.
+    - Type: get_text
+      Strategy: xpath
+      Id: $PAGE.care_platform_participant.first_darkened_letter_button$
```
Comment (csanchez-nts) [inline tests/cases/providers_care_platform/case_providers_participants.yaml:19]: same here

**Review by csanchez-nts — CHANGES_REQUESTED** [review]
Update all page object references to be from provider page object not care partner platform

File: tests/cases/providers_care_platform/case_providers_participants.yaml (hunk 2 above)
Comment (cmoreno-nts) [reply tests/cases/providers_care_platform/case_providers_participants.yaml]: Done.

File: tests/cases/providers_care_platform/case_providers_participants.yaml (hunk 3 above)
Comment (cmoreno-nts) [reply tests/cases/providers_care_platform/case_providers_participants.yaml]: Done.

File: tests/page_objects/providers_care_platform_page.yaml [hunk 4]
```diff
@@ -39,21 +40,26 @@ providers_facilities_page:
...
+  devices_button: //button[text()='Devices']
+  tabs_content: //div[@class='tabs-content']
+  title_subtab_devices: //h4[text()='Facility Devices']
   call_based_on_device_button: //strong[text()='Device $AND_CLI_facility_devices_user1_number$:']/ancestor::tr//a[@class='button button-basic text-xs']
   message_based_on_device_button: //strong[text()='Device $AND_CLI_facility_devices_user1_number$:']/ancestor::tr//button[@class='button button-basic text-xs']
   message_input: //input[@id='message']
   send_message_button: //span[@class='icon-send text-md']/ancestor::button
   last_message_sent: (//div[@class='message message-to message-to-inverted'])[last()]
+  info_title: //h4[text()='Facility Information']
+  device_status: //div[@class='account-status account-status-active']
```
[the remark this answers is not in the document]
Comment (cmoreno-nts) [reply tests/page_objects/providers_care_platform_page.yaml]: Done

File: tests/page_objects/providers_care_platform_page.yaml [hunk 5]
```diff
@@ -39,27 +40,32 @@ providers_facilities_page:
   facilities_header_title: //h3
   alpha_center_td: //div[@role='button']//div[text()='Alpha Healthcare Center']
   call_portal_btn: //button[text()='Call Portal']
+  records_button: //a/button[text()='Records']
```
Comment (csanchez-nts) [inline tests/page_objects/providers_care_platform_page.yaml:43]: this is a tab not a button

**Review by csanchez-nts — APPROVED** [review]

**Review by miguel-hquez — APPROVED** [review]

**Pull request merged**